### PR TITLE
Add encryption in S3 buckets

### PR DESCRIPTION
**Pull request by pipo02mix**
By default encrypt S3 buckets.

### CODE REVIEW

**Comment by pipo02mix** [comment]
Tested on ginger (current and upgrade from last version)

File: service/controller/v25/key/key.go [hunk 1]
```diff
@@ -47,6 +47,9 @@ const (
 	// LogDeliveryURI is used for setting the correct ACL in the access log bucket
 	LogDeliveryURI = "uri=http://acs.amazonaws.com/groups/s3/LogDelivery"
 
+	// S3BucketEncryptionAlgorithm is used to determine which algorithm use S3 to encrypt buckets.
+	S3BucketEncryptionAlgorithm = "AES256"
```
Comment (xh3b4sd) [inline service/controller/v25/key/key.go]: The `key` package is overloaded with all kinds of information not relevant for multiple packages. Let's keep the keys and constants relevant for a single package in the package that is interested in the information. That way the key package should become leaner and only hold information relevant in multiple places.

Comment (pipo02mix) [reply service/controller/v25/key/key.go]: ok so I create a new key package under the resource? or do I include the key in same file (resource/create.go)?

Comment (xh3b4sd) [reply service/controller/v25/key/key.go]: I think it is fine to just put it into the file where it is used.

**Comment by xh3b4sd** [comment]
I asked a couple of questions above to better understand the mechanism here. Can somebody answer them?

**Comment by mazzy89** [comment]
@xh3b4sd I can address your questions since are related to AWS implementation. The encryption chosen here is the default S3 server-side encryption provided by S3 as here explained https://docs.aws.amazon.com/AmazonS3/latest/user-guide/default-bucket-encryption.html and https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html

We might here also consider enforcing encryption through bucket policy. Not sure if we use bucket policy though. I'll go to check it.

**Comment by tuommaki** [comment]
> We might here also consider enforcing encryption through bucket policy. Not sure if we use bucket policy though. I'll go to check it.

I somehow was under the impression that the AWS SDK actually does this behind the scenes, but this is not the case?

**Comment by mazzy89** [comment]
I was wrong @tuommaki I've checked it better. When the default encryption is enabled then all the `PUT` operations against the bucket are executed by passing in the correct headers to encrypt the uploaded object in the bucket. so all good.

**Comment by pipo02mix** [comment]
> I asked a couple of questions above to better understand the mechanism here. Can somebody answer them?

Sorry I missed the message questions, only saw the code ones.

Thanks Salvo for replying. Yes I checked and it works fine, AWS take care of the encryption at rest.

**Comment by pipo02mix** [comment]
@xh3b4sd is fine for you?

**Pull request merged**